cabal-install/Main.hs

```diff
@@ -258,6 +256,7 @@ mainWorker args = topHandler $
       , regularCmd runCommand runAction
       , regularCmd initCommand initAction
       , regularCmd configureExCommand configureAction
+      , regularCmd reconfigureCommand reconfigureAction
```

Comment:
@ttuegel Yes, it'd be nice with some more documentation describing the use cases. In my understanding, since reconfiguring normally happens automatically, users should almost never have to run `reconfigure` manually.